File: class_7/class_7/app/src/main/java/com/alejouribesanchez/class_7/Main3Activity.java
package com.alejouribesanchez.class_7;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class Main3Activity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main3);

        Button button = (Button) findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(Main3Activity.this,Main4Activity.class));
            }
        });

        View layout = (View)findViewById(R.id.activity_main3);
        layout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(Main3Activity.this, Main4Activity.class));
            }
        });
    }
}

File: class_7/class_7/app/src/main/java/com/alejouribesanchez/class_7/Main3Activity.kt
package com.alejouribesanchez.class_7

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class Main3Activity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main3)

        val button = findViewById(R.id.button) as Button
        button.setOnClickListener { startActivity(Intent(this@Main3Activity, Main4Activity::class.java)) }

        findViewById(R.id.activity_main3).setOnClickListener { startActivity(Intent(this@Main3Activity, Main4Activity::class.java)) }
    }
}
